refinery/data_set_manager/tests.py

```diff
@@ -1966,32 +1978,40 @@ def unsuccessful_import_assertions(self):
         self.assertEqual(Investigation.objects.count(), 0)
         self.assertEqual(Assay.objects.count(), 0)
 
+
+class ProcessISATabViewTests(ProcessISATabViewTestBase):
     @mock.patch.object(data_set_manager.views.import_file, "delay")
     def test_post_good_isa_tab_file(self, delay_mock):
         with open('data_set_manager/test-data/rfc-test.zip') as good_isa:
-            self.client.post(
-                self.isa_tab_import_url,
-                data={
-                    "isa_tab_url": None,
-                    "isa_tab_file": good_isa
-                },
-                HTTP_X_REQUESTED_WITH='XMLHttpRequest'
-            )
+            self.post_isa_tab(isa_tab_file=good_isa)
         self.successful_import_assertions()
 
     def test_post_bad_isa_tab_file(self):
         with open('data_set_manager/test-data/HideLabBrokenA.zip') as bad_isa:
-            self.client.post(
-                self.isa_tab_import_url,
-                data={
-                    "isa_tab_url": None,
-                    "isa_tab_file": bad_isa
-                },
-                HTTP_X_REQUESTED_WITH='XMLHttpRequest'
-            )
+            self.post_isa_tab(isa_tab_file=bad_isa)
+        self.unsuccessful_import_assertions()
+
+    def test_post_bad_isa_tab_url(self):
+        self.post_isa_tab(isa_tab_url="non-existant-file")
         self.unsuccessful_import_assertions()
 
 
+class ProcessISATabViewLiveServerTests(ProcessISATabViewTestBase,
+                                       LiveServerTestCase):
+    @mock.patch.object(data_set_manager.views.import_file, "delay")
+    def test_post_good_isa_tab_url(self, delay_mock):
+        original_media_root = settings.MEDIA_ROOT
+        settings.MEDIA_ROOT = os.path.join(
+            settings.BASE_DIR,
+            "refinery/data_set_manager/test-data/"
+        )
+        media_url = urljoin(self.live_server_url, settings.MEDIA_URL)
+        good_isa_tab_url = urljoin(media_url, "rfc-test.zip")
+        self.post_isa_tab(isa_tab_url=good_isa_tab_url)
+        self.successful_import_assertions()
+        settings.MEDIA_ROOT = original_media_root
```

Comment:
If one of the assertions fails, this won't be reset, and we'd get more failures downstream? maybe it belongs in teardown?

Reply:
After some digging I found [SimpleTestCase.settings context manager](https://docs.djangoproject.com/en/1.7/topics/testing/tools/#django.test.SimpleTestCase.settings) I'm gonna try it out.